### ISSUE TITLE
VS Code: Refactor does not appear on editor context menu

### ISSUE THREAD

**Issue by andim27** [dart-lang/sdk#48224]
VS Code ver 1.63
Flutter (Channel stable, 2.8.1, on Linux Mint 20.3 5.4.0-96-generic, locale en_US.UTF-8)
flutter doctor - all is ok
![vs_code_no_error](https://user-images.githubusercontent.com/4599713/151164735-8dba33cb-a036-4473-8543-f2ad90d24ff8.png)
![vs_code_refactoring_error](https://user-images.githubusercontent.com/4599713/151164743-f353964f-bfb4-47f3-9a22-9f4df8365ba8.png)

**Comment by bwilkerson** [comment]
@DanTup

**Comment by DanTup** [comment]
@andim27 in one of your screenshots the items is there, and in the other it's not. What did you do differently between the screenshots?

It looks like perhaps the extension has not been activated for the second screenshot. Do you see any errors listed if you click Help -> Toggle Developer Tools in VS Code, then click on the Console tab at the top?

**Comment by andim27** [comment]
This has been happening quite often lately when building large projects.

**Comment by andim27** [comment]
yes, sometimes the PopUp Window from analysis server appears but restarting it doesn't helps

**Comment by DanTup** [comment]
@andim27 do you know exactly what the popup says? Can you give a screenshot next time it occurs? Could you also check the console in the VS Code developer tools mentioned above to see if there is a more detailed error there when this happens? Thanks!

**Comment by andim27** [comment]
{"method":"$/progress","params":{"token":"ANALYZING","value":{"kind":"end"}},"jsonrpc":"2.0"}
[12:47:57 PM] [Analyzer] [Info] <== Content-Length: 220
Content-Type: application/vscode-jsonrpc; charset=utf-8
[12:48:12 PM] [Analyzer] [Error] ../../runtime/vm/zone.cc: 94: error: Out of memory.
version=2.15.1 (stable) (Tue Dec 14 13:32:21 2021 +0100) on "linux_x64"
pid=8191, thread=12640, isolate_group=(nil)((nil)), isolate=(nil)((nil))
isolate_instructions=0, vm_instructions=559bdc0c5600
[12:48:12 PM] [Analyzer] [Error] ../../runtime/vm/zone.cc: 94: error: Out of memory.
Aborting re-entrant request for stack trace.
[12:48:40 PM] [Analyzer] [Error] Aborted (core dumped)
[12:48:40 PM] [Analyzer] [Info] Spawning.../bin/cache/dart-sdk/bin/dart with args ["language-server","--client-id=VS-Code","--client-version=3.32.0"]
[12:48:40 PM] [Analyzer] [Info]     PID: 12894
[12:48:41 PM] [Analyzer] [Info] ==> Content-Length: 4917

**Comment by DanTup** [comment]
@andim27 could you post the full output of running `code --status` when you have the project open that's causing this (ideally after you've been coding for some time, so if it's memory usage is growing that may be clear from the stats)? Thanks!

**Comment by andim27** [comment]
Now - not but in any cases dart-tooling, refactoring in the right click menu is needed to be more flexible.It's very slowly and its own life(((

**Comment by DanTup** [comment]
@andim27 do you mean the crash is gone and the refactors are visible, but they take a long time to execute?

If so, does this happen always, or is it intermittent?

**Comment by andim27** [comment]
this cases appears on large projects
